src/core/builders.ts

```diff
@@ -80,6 +80,23 @@ export interface PayloadBuilder {
    */
   nonce(nonce: number): NonNil<PayloadBuilder>;
 
+
+  /**
+   * Challenge for the transaction.
+   *
+   * @param {string} challenge- The challenge for the transaction.
+   * @returns {PayloadBuilder} The current `PayloadBuilder` instance for chaining.
+   */
+  challenge(challenge: Nillable<string>): NonNil<PayloadBuilder>;
+
+  /**
+   * Signature for the transaction.
+   *
+   * @param {string} signature- The signature for the transaction.
```

Comment:
This isn't the signature for the transaction. It is the signature for the call request. Transaction signatures are for when you want to execute a state change on Kwil, which is different.

Transaction signatures are created inside the builders. This method is for a signature that is used to authenticate for call requests.